src/main/kotlin/org/rust/cargo/toolchain/Rustfmt.kt

```diff
@@ -55,11 +57,9 @@ class Rustfmt(private val rustfmtExecutable: Path) {
         }
 
         val processOutput = try {
-            GeneralCommandLine(rustfmtExecutable)
+            toolchain.createGeneralCommandLine(RUSTFMT, *arguments.toTypedArray())
                 .withWorkDirectory(cargoProject.workingDirectory)
-                .withParameters(arguments)
-                .withCharset(Charsets.UTF_8)
-                .execute(cargoProject.project, false, stdIn = document.text.toByteArray())
+                .execute(Disposer.newDisposable("Rustfmt"), false, stdIn = document.text.toByteArray())
```

Comment:
Would it be okay to use `project.taskQueue` as the disposable, seeing as it is a project level `Service`? I'm asking because [the documentation](https://jetbrains.org/intellij/sdk/docs/basics/disposers.html#choosing-a-disposable-parent) says that using the `project` as a `Disposable` will cause a memory leak.

Reply:
A memory leak is true for unloadable plugins. Our plugin is not unloadable (unloading is explicitly forbidden), so it's OK to use Project as Disposable and it's ok to ignore the warning and the documentation paragraph, please use Project for now.